alns/stop/NoImprovement.py

```diff
@@ -10,25 +10,22 @@ class NoImprovement(StoppingCriterion):
     Criterion that stops if the best solution has not been improved
     after a number of iterations.
 
-    The first iteration is used to initialize the criterion. Hence,
-    the algorithm will iterate at least n_iterations + 1 times.
-
     Parameters
     ----------
-    n_iterations
+    max_iterations
         The maximum number of non-improving iterations.
     """
 
-    def __init__(self, n_iterations: int):
-        if n_iterations < 0:
-            raise ValueError("n_iterations < 0 not understood.")
+    def __init__(self, max_iterations: int):
+        if max_iterations < 0:
+            raise ValueError("max_iterations < 0 not understood.")
 
-        self._n_iterations = n_iterations
+        self._max_iterations = max_iterations
         self._target: Optional[float] = None
```

Comment:
Issue: AFAICT `self._counter` is never initialised before `__call__`.

Reply:
[I had expected mypy to flag this (since we're adding an impromptu field not declared in `__init__`, but perhaps I have missed something]

Reply:
> [I had expected mypy to flag this (since we're adding an impromptu field not declared in `__init__`, but perhaps I have missed something]

I expected that as well, but running mypy didn't flag anything.